third-party/cmake/DownloadAndExtractFile.cmake

```diff
@@ -85,6 +75,25 @@ if(NOT status_code EQUAL 0)
 ")
 endif()
 
+set(NULL_SHA1 "0000000000000000000000000000000000000000")
+
+# We could avoid computing the SHA1 entirely if a NULL_SHA1 was given,
+# but we want to warn users of an empty file.
+file(SHA1 ${file} ACTUAL_SHA1)
+if(ACTUAL_SHA1 STREQUAL "da39a3ee5e6b4b0d3255bfef95601890afd80709")
+  # File was empty.  It's likely due to lack of SSL support.
+  message(FATAL_ERROR
+    "Failed to download ${URL}.  The file is empty and likely means CMake "
+    "was built without SSL support.  Please use a version of CMake with "
+    "proper SSL support.")
```

Comment:
I don't see that link offering anything the message doesn't.  If there was more there that would help with this specific problem it would seem worthwhile, but there's not (other than a link to a ticket, which says the same thing essentially).